src/analysis/convex/basic.lean

```diff
@@ -1035,21 +1035,18 @@ end convex_hull
 
 section simplex
 
-variables (ι : Type*) [fintype ι]
+variables (ι : Type*) [ordered_semiring 𝕜] [fintype ι]
 
-/-- The standard simplex in the space of functions `ι → ℝ` is the set
-of vectors with non-negative coordinates with total sum `1`. -/
-def std_simplex (ι : Type*) [fintype ι] (R : Type*) [ordered_semiring R] :
-  set (ι → R) :=
+/-- The standard simplex in the space of functions `ι → 𝕜` is the set of vectors with non-negative
+coordinates with total sum `1`. This is the free object in the category of convex spaces.-/
+def std_simplex : set (ι → 𝕜) :=
```

Comment:
Does this generalisation of `std_simplex` have to happen simultaneously with the headline item of generalising convexity (concavity) of functions?

Reply:
No, it is a reliquate from #9268 that also happens to be quite small.

Reply:
In that case, I think these two pieces should be PR'd separately.

I'd expect that the generalisation of `std_simplex` makes sense on its own.